drivers/sensor/tdk/icm42670/Kconfig

```diff
@@ -9,13 +10,44 @@ menuconfig ICM42670
 	bool "ICM42670 Six-Axis Motion Tracking Device"
 	default y
 	depends on DT_HAS_INVENSENSE_ICM42670_ENABLED
+	select ZEPHYR_HAL_TDK_MODULE
 	select SPI if $(dt_compat_on_bus,$(DT_COMPAT_INVENSENSE_ICM42670),spi)
 	select I2C if $(dt_compat_on_bus,$(DT_COMPAT_INVENSENSE_ICM42670),i2c)
 	help
-	  Enable driver for ICM42670 SPI-based six-axis motion tracking device.
+	  Enable driver for ICM42670 SPI-based or I2C-based Six-Axis Motion Tracking device.
 
 if ICM42670
 
+choice ICM_VARIANT
+	prompt "ICM variant to identify the sensor"
+	help
+	  Select the ICM variant
+config ICM42670P
+	bool "ICM42670-P IMU 6-Axis Motion Tracking"
+config ICM42670S
+	bool "ICM42670-S IMU 6-Axis Motion Tracking with Air Motion Library (AML)"
+endchoice
```

Comment:
This should go to device tree instead of Kconfig, imagine having multiple variants.

Reply:
Thanks for this comment as well. I did the change and put this option in device tree. It has more sense like this.